### PR TITLE
AMDGPU: update GFX11 wmma hazards

### PR DESCRIPTION
**Pull request by petar-avramovic**
One V_NOP or unrelated VALU instruction in between is required for correctness when matrix A or B of current WMMA instruction overlaps with matrix D of previous WMMA instruction.
Remaining cases of WMMA operand overlaps are handled by the hardware and do not require handling in hazard recognizer.

Hardware may stall in cases where:
- matrix C of current WMMA instruction overlaps with matrix D of previous WMMA instruction
- VALU instruction reads matrix D of previous WMMA instruction
- matrix A,B or C of WMMA instruction reads result of previous VALU instruction

### CODE REVIEW

**Comment by petar-avramovic** [comment]
ping

File: llvm/lib/Target/AMDGPU/GCNHazardRecognizer.cpp [hunk 1]
```diff
@@ -1728,25 +1728,6 @@ bool GCNHazardRecognizer::fixWMMAHazards(MachineInstr *MI) {
...
-      const MachineOperand *Src2Mods =
-          TII->getNamedOperand(*MI, AMDGPU::OpName::src2_modifiers);
-      const bool NoSrc2Mods =
-          (Src2Mods->getImm() & (SISrcMods::NEG | SISrcMods::NEG_HI)) == 0;
-      // Exception: there is no hazard if the wmma instructions are of the same
-      // type and there is no input modifier on src2 of the current instruction.
-      return !(NoSrc2Mods && (TII->pseudoToMCOpcode(I.getOpcode()) ==
-                              TII->pseudoToMCOpcode(MI->getOpcode())));
-    }
-
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/GCNHazardRecognizer.cpp:1749]: How did this end up here in the first place?

Comment (piotrAMD) [reply llvm/lib/Target/AMDGPU/GCNHazardRecognizer.cpp:1749]: That was coded according to the spec, but the behavior/spec have recently been clarified: inserting V_NOP in these cases won't help correctness and performance at all.

**Review by piotrAMD — APPROVED** [review]

**Comment by jplehr** [comment]
This patch resulted in two additional failures in AMDGPU OpenMP Offloading bot.
https://lab.llvm.org/staging/#/builders/185/builds/744

**Pull request merged**